### PR TITLE
Fast follow: post parent ID cache V2

### PR DESCRIPTION
**Pull request by spacedmonkey**
PR is based on https://github.com/WordPress/wordpress-develop/pull/5386. It is simplier PR, that just changes the function name and prime the post parent cache on post save. 

Trac ticket: https://core.trac.wordpress.org/ticket/59188

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/post.php [hunk 1]
```diff
@@ -3559,6 +3559,22 @@ function _reset_front_page_settings_for_post( $post_id ) {
...
+ * The post parent cache is used to store the parent ID of a post for quick retrieval.
+ *
+ *  @since 6.4.0
+ *  @access private
+ *
+ * @param int    $post_id The ID of the post for which to set or update the parent cache.
+ * @param WP_Post $post   The post object representing the post.
+ */
+function _set_post_parent_cache( $post_id, $post ) {
+	wp_cache_set( $post_id, $post->post_parent );
```
Comment (peterwilsoncc) [inline src/wp-includes/post.php]: Group missing but see main comment.

```suggestion
	wp_cache_set( $post_id, $post->post_parent, 'post_parent' );
```

File: src/wp-includes/post.php [hunk 2]
```diff
@@ -3559,6 +3559,22 @@ function _reset_front_page_settings_for_post( $post_id ) {
...
 }
 
+/**
+ * Set or update the post parent cache for a specific post.
+ *
+ * This function sets or updates the post parent cache for a given post ID.
+ * The post parent cache is used to store the parent ID of a post for quick retrieval.
+ *
+ *  @since 6.4.0
+ *  @access private
```
Comment (peterwilsoncc) [inline src/wp-includes/post.php:3569]: I think this can be public as with the other cache warming functions.

**Review by peterwilsoncc — COMMENTED** [review]
Added a few notes.

Something that occurred to me over the weekend is that the second group may cause problems. Until now running `wp_cache_flush_group( 'posts' )` would clear all the post object related data but with this change a developer now needs to flush the new group too.

Unlike the recent change to introduce the `post-queries` group, I'm not sure the `post_parent` group provides enough benefit for back-compat break as the individual post objects do not use the last changed date filling persistent caches with unreachable keys.

**Comment by spacedmonkey** [comment]
```wp_cache_flush_group( 'posts' )```

This is a documented way of clearing caches. You should call `clean_post_cache` is the way of cleaning caches. 

We could reuse the cache group post and make the id something like this `post_parent-123`

**Comment by spacedmonkey** [comment]
Closing in favour of https://github.com/WordPress/wordpress-develop/pull/5442 and https://github.com/WordPress/wordpress-develop/pull/5443

**Pull request closed without merging**